src/dodal/devices/pgm.py

```diff
@@ -0,0 +1,41 @@
+from enum import Enum
+
+from ophyd_async.core import (
+    ConfigSignal,
+    StandardReadable,
+)
+from ophyd_async.epics.motor import Motor
+from ophyd_async.epics.signal import epics_signal_rw
+
+
+class PGM(StandardReadable):
+    """
+    Plane grating monochromator, it is use in soft x-ray beamline to generate monochromic beam.
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        grating: type[Enum],
```

Comment:
could: replace with a direct reference to I10Grating class? later on could be added `or OtherGrating` ?